src/machine_charm.py

```diff
@@ -47,6 +47,12 @@ def __init__(self, *args):
         self.framework.observe(self.on.remove, self.on_remove)
         self.framework.observe(self.on["juju_info"].relation_joined, self._update_config)
 
+        # TODO observe cos_machine changed and call e.g.
```

Comment:
In the end, the machine charm will be able to just call `self._update_config(event)`, basically. 

The logic in `_update_status` is somewhat too dependent on the behavior of the k8s version, since the machine charm probably _always_ wants a remote write but doesn't care about a metrics endpoint, but that's for a different patch.

Either way, both log target changes and scrape changes will want to call `update_config`, so may as well just clal that endpoint for every machine event.